src/application.rs

```diff
@@ -18,69 +18,80 @@ pub trait ApplicationHandler {
+    /// **Android / macOS / Orbital / Wayland / Windows / X11:** Unsupported.
     ///
-    /// [`applicationDidBecomeActive`]: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622956-applicationdidbecomeactive
-    /// [iOS application lifecycle]: https://developer.apple.com/documentation/uikit/app_and_environment/managing_your_app_s_life_cycle
+    /// [`resumed()`]: Self::resumed
+    fn resumed(&mut self, event_loop: &ActiveEventLoop) {
+        let _ = event_loop;
+    }
+
+    /// Emitted from the point onwards the application should create render surfaces.
     ///
-    /// ## Web
+    /// See [`destroy_surfaces()`].
     ///
-    /// On Web, the `Resumed` event is emitted in response to a [`pageshow`] event
-    /// with the property [`persisted`] being true, which means that the page is being
-    /// restored from the [`bfcache`] (back/forward cache) - an in-memory cache that
-    /// stores a complete snapshot of a page (including the JavaScript heap) as the
-    /// user is navigating away.
+    /// ## Portability
     ///
-    /// [`pageshow`]: https://developer.mozilla.org/en-US/docs/Web/API/Window/pageshow_event
-    /// [`persisted`]: https://developer.mozilla.org/en-US/docs/Web/API/PageTransitionEvent/persisted
-    /// [`bfcache`]: https://web.dev/bfcache/
-    /// [`Suspended`]: Self::suspended
-    fn resumed(&mut self, event_loop: &ActiveEventLoop);
+    /// It's recommended that applications should only initialize their render surfaces after the
+    /// [`can_create_surfaces()`] method is called. Some systems (specifically Android) won't allow
+    /// applications to create a render surface until that point.
+    ///
+    /// For consistency, all platforms call this method even if they don't themselves have a formal
+    /// surface destroy/create lifecycle. For systems without a surface destroy/create lifecycle the
+    /// [`can_create_surfaces()`] event is always emitted after the [`StartCause::Init`] event.
+    ///
+    /// Applications should be able to gracefully handle back-to-back [`can_create_surfaces()`] and
+    /// [`destroy_surfaces()`] calls.
+    ///
+    /// ## Platform-specific
+    ///
+    /// ### Android
+    ///
+    /// On Android, the [`can_create_surfaces()`] method is called when a new [`SurfaceView`] has
+    /// been created. This is expected to closely correlate with the [`onResume`] lifecycle
+    /// event but there may technically be a discrepancy.
+    ///
+    /// [`onResume`]: https://developer.android.com/reference/android/app/Activity#onResume()
+    ///
+    /// Applications that need to run on Android must wait until they have been "resumed" before
+    /// they will be able to create a render surface (such as an `EGLSurface`, [`VkSurfaceKHR`]
+    /// or [`wgpu::Surface`]) which depend on having a [`SurfaceView`]. Applications must also
+    /// assume that if they are [suspended], then their render surfaces are invalid and should
+    /// be dropped.
```

Comment:
Now that the wording is fixed, I don't think this makes any sense: your render surface becomes invalid when Android says so; don't assume this when another event is received.

Reply:
I'm not following, what do you mean with "another event"?
The word "assume" should probably be removed here?

Reply:
It should just be removed altogether:

1. Android will raise an event when your surface is destroyed;
2. Hence when you destroy it in `suspended()` too, you now have to add complicated extra logic to make sure it's only destroyed in one of the two paths;
3. When an Android app is suspended in the [`onPause()`](https://developer.android.com/guide/components/activities/activity-lifecycle#onpause) sense (pending how #3779 pans out), the app _might still be visible_ and should have a render surface...

Reply:
Addressed in #3786 by replacing `suspended` with `onStart`/`onStop` and merely direct users to the `destroy_surfaces()` callback which is the only valid and consistent place.